snuba/query/snql/__init__.py

```diff
@@ -1,23 +1,280 @@
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    List,
+    NamedTuple,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
 from parsimonious.grammar import Grammar
+from parsimonious.nodes import Node, NodeVisitor
+
 from snuba.datasets.dataset import Dataset
-from snuba.query.logical import Query
+from snuba.query.dsl import divide, minus, multiply, plus
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
+from snuba.query.logical import Query, SelectedExpression
+
+
+snql_grammar = Grammar(
+    fr"""
+    query_exp             = match_clause? where_clause? collect_clause? having_clause? order_by_clause?
+    match_clause          = space* "MATCH" space* "(" clause ")" space*
+    where_clause          = space* "WHERE" clause space*
+    collect_clause        = space* "COLLECT" collect_list space*
+    order_by_clause       = space* "ORDER BY" collect_list ("ASC"/"DESC") space*
+
+    collect_list          = collect_columns* (low_pri_arithmetic)
```

Comment:
Maybe you can just move the content of this one directly into the `collect_clause` above? It does not seem it needed to be split out in its own rule.

Reply:
I am still not sure I get why you need to split `collect_list` from `collect_columns`. Can't `collect_list` just be `(low_pro_arithmetic space* comma space*)* low_pri_arithmetic` ?